docs/proposals/policy.md

```diff
@@ -0,0 +1,270 @@
+```
+{
+	"kind": "role",
+	"name": "view",
+	"namespace": "global",
+	"rules": [
+		{ "verbs": ["watch", "list", "get"], "resourceKinds": ["*"] },
+	]
+}
+{
+	"kind": "role",
+	"name": "edit",
+	"namespace": "global",
+	"rules": [
+		{ "verbs": ["*"], "resourceKinds": ["*"] }
+		{ "deny": "true", "verbs": ["create", "update", "delete"], "resourceKinds": ["PolicyRuleBinding", "Role", RoleBinding"] }
+	]
+}
+{
+	"kind": "role",
+	"name": "admin",
+	"namespace": "global",
+	"rules": [
+		{ "verbs": ["*"], "resourceKinds": ["*"] }
+	]
+}
+```
+
+The global admin (guy who owns the openshift installation), creates a project `chisel` and assigns `Hubert` as the admin for the project using the predefined role.  The permissions are expressed like this:
+```
+{
+	"kind": "roleBinding",
+	"name": "ProjectAdmins",
+	"namespace": "chisel",
+	"referencesGlobalRole": "true",
+	"roleName": "admin",
+	"userNames": ["Hubert"]
+}
+```
+
+`Hubert` now has the power to manage policy on the `chisel` project.  He can grant permission to users who should have access.  In this case, he grants edit permission to `Curt` using the predefined role.
+```
+{
+	"kind": "roleBinding",
+	"name": "Editors",
+	"namespace": "chisel",
+	"referencesGlobalRole": "true",
+	"roleName": "editor",
+	"userNames": ["Curt"]
+}
+```
+
+That covers most cases where global admins want to create projects and delegate the administration of a project to a project admin.  Then the project admin wants to delegate project level permissions to other users.  We can continue the example with `Hubert` (our project admin) wanting to create custom roles and restrictions.
+
+`Hubert` needs wants to create a bot that only has permission to touch labels on DeploymentConfigs.  He can create a role like this (note that a role is required because we want to have a set of two rules):
+```
+{
+	"kind": "role",
+	"name": "deploymentConfigLabelers",
+	"namespace": "chisel",
+	"rules": [
+		{ "verbs": ["watch", "list", "get"], "resourceKinds": ["DeploymentConfig"] },
+		{ "verbs": ["update"], "resourceKinds": ["DeploymentConfig"], "attributeRestrictions" : {"fieldsMutatable": ["labels"]} }
+	]
+}
+{
+	"kind": "roleBinding",
+	"name": "DeploymentConfigLabelerBots",
+	"namespace": "chisel",
+	"roleName": "deploymentConfigLabelers",
+	"userNames": ["ProtectorBot", "DeprotectorBot"]
+}
+```
+
+Finally, `Hubert` wants to prevent `Curt` from ever deleting a protected deploymentConfig again, so he creates this policyRuleBinding:
+```
+{
+	"kind": "policyRuleBinding",
+	"name": "FatFingeredEditors",
+	"namespace": "chisel",
+	"rule": { "Deny": "true", "Verbs": ["delete"], "ResourceKinds": ["DeploymentConfig"], "AttributeRestrictions" : {"labelsContain": ["protected"]} }
```

Comment:
any reason to not let this express multiple allow/denies the same way a role does?